File: JavaScript/2nd_ex/main.js
function err(exc) {
    throw new Error(exc);
}

function init() {
    const val = parseInt(prompt());
    if (isNaN(val) || val <= 0) {
        err("Invalid input");
    }
    return val;
}

function cmp(arr, v) {
    if (v < Math.max(...arr)) {
        err("The volume of the watering can is not enough to water the flower");
    }
}

function countingSteps(arr, v) {
    let count = 0;
    let tmp = v;

    for (let i = 0; i < arr.length; i++) {
        if (tmp < arr[i]) {
            tmp = v;
            count += i * 2;
        }
        count += 1;
        tmp -= arr[i];
    }

    return count;
}
try {
    const n = init();

    const arr = [];
    for (let i = 0; i < n; i++) {
        arr.push(init());
    }

    const v = init();

    cmp(arr, v);
    
    console.log(countingSteps(arr, v));
} catch (e) {
    console.error(e.message);
}

File: kotlin/2nd_ex/main.kt
fun main() {
    val n = init()
    val arr = mutableListOf<Int>()
    repeat(n) {
        arr.add(init())
    }
    val v = init()

    cmp(arr, v)
    println(countingSteps(arr, v))
}

fun err(exc: String): Nothing {
    throw Exception(exc)
}

fun init(): Int {
    val valStr = readLine() ?: err("Invalid input")
    val value = valStr.toInt()
    if (value <= 0) err("Invalid input")
    return value
}

fun cmp(arr: List<Int>, v: Int) {
  val max = arr.max() ?: err("Invalid input")
  if (v < max) {
    err("The volume of the watering can is not enough to water the flower")
  }
}

fun countingSteps(arr: List<Int>, v: Int): Int {
    var count = 0
    var tmp = v

    arr.forEachIndexed { i, value ->
        if (tmp < value) {
            tmp = v
            count += i * 2
        }
        count++
        tmp -= value
    }
    return count
}
